### ISSUE TITLE
Bad parsing behavior for generic arrow type arguments

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#23996]
```ts
// No parse error.
type Foo = ReturnType< <T>(x: T) => T>;

// Parse error.
type Bar = ReturnType<<T>(x: T) => T>;
```

**Expected**: No parse error, `Foo` and `Bar` are equivalent.
**Actual**: Parse errors, `Bar` ends up as nonsense (`any`) because of a parse error.

**Comment by DanielRosenwasser** [comment]
We're probably parsing `<<` as a left shift
